### ISSUE TITLE
UIViewController `viewIsAppearing` is not supported by `type_contents_order` rule

### ISSUE THREAD

**Issue by way-to-code** [realm/SwiftLint#5259]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Since Xcode 15 SDK Apple introduced a new `UIViewController` life cycle method `viewIsAppearing(_:)`. The official documentation for the new method [can be found here](https://developer.apple.com/documentation/uikit/uiviewcontroller/4195485-viewisappearing).

As for now `type_contents_order` rule considers the new method as `other_method` that is not correct. The new method should be considered as `view_life_cycle_method`.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'ViewController.swift' (1/3)
ViewController.swift:15:14: warning: Type Contents Order Violation: An 'other_method' should not be placed amongst the type content(s) 'view_life_cycle_method' (type_contents_order)
Done linting! Found 1 violation, 0 serious in 3 files.
```

### Environment

* SwiftLint version: 0.53.0
* Installation method used: pkg file
* Configuration file:

```yml
only_rules:
    - type_contents_order
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
 No

* Which Xcode version are you using (check `xcodebuild -version`)?
Xcode 15.0 Build version 15A240d

* Do you have a sample that shows the issue?

    ```swift
    final class ViewController: UIViewController {
        override func viewDidLoad() {
            super.viewDidLoad()
        }
        
        override func viewIsAppearing(_ animated: Bool) {
            super.viewIsAppearing(animated)
        }
        
        override func viewDidAppear(_ animated: Bool) {
            super.viewDidAppear(animated)
        }
    }
    ```
